client/src/components/profile/Profile.js

```diff
@@ -128,43 +128,73 @@ function Profile({ user, isSessionUser }) {
   if (isLocked) {
     return renderIsLocked(username);
   }
+
   return (
-    <Fragment>
-      <Helmet>
-        <title>Profile | freeCodeCamp.org</title>
-      </Helmet>
-      <Spacer size={2} />
-      <Grid>
-        {isSessionUser ? renderSettingsButton() : null}
-        <Camper
-          about={showAbout && about}
-          githubProfile={githubProfile}
-          isGithub={isGithub}
-          isLinkedIn={isLinkedIn}
-          isTwitter={isTwitter}
-          isWebsite={isWebsite}
-          linkedin={linkedin}
-          location={showLocation && location}
-          name={showName && name}
-          picture={picture}
-          points={showPoints && points}
-          twitter={twitter}
-          username={username}
-          website={website}
-          yearsTopContributor={yearsTopContributor}
-        />
-        {showHeatMap ? <HeatMap calendar={calendar} streak={streak} /> : null}
-        {showCerts ? <Certifications username={username} /> : null}
-        {showPortfolio ? <Portfolio portfolio={portfolio} /> : null}
-        {showTimeLine ? (
-          <Timeline
-            className='timelime-container'
-            completedMap={completedChallenges}
-            username={username}
-          />
-        ) : null}
-      </Grid>
-    </Fragment>
+    <StaticQuery
+      query={graphql`
+        query ProfileQuery {
+          allChallengeNode(
+            sort: { fields: [superOrder, order, challengeOrder] }
+          ) {
+            edges {
+              node {
+                fields {
+                  slug
+                }
+                id
+              }
+            }
+          }
+        }
+      `}
+      render={data => {
+        const idToSlugMap = data.allChallengeNode.edges.reduce(
+          (a, b) => Object.assign(a, { [b.node.id]: b.node.fields.slug }),
+          {}
+        );
```

Comment:
I do not love how idToSlugMap is made. there must be a better way using graphql. please let me know if anything comes to mind.